ledger/store/src/helpers/mod.rs

```diff
@@ -61,7 +61,11 @@ macro_rules! atomic_batch_scope {
             },
             // Rewind this atomic batch scope.
             Err(err) => {
-                $self.atomic_rewind();
+                if is_atomic_in_progress {
```

Comment:
Could you add documentation for the cases here?

Reply:
Also, how often is this case actual hit in our code?

When do we perform `atomic_batch_scope` without `atomic_finalize` being called first?